sw/device/tests/chip_power_idle_load.c

```diff
@@ -0,0 +1,516 @@
+
+  // Set output modes of all GPIO pins
+  CHECK_DIF_OK(dif_gpio_output_set_enabled_all(&gpio, kGpioMask));
+
+  // Write to set IOA2 low at the start of the test:
+  CHECK_DIF_OK(dif_gpio_write(&gpio, 2, 0));
+
+  // ------------------------------------
+  // RV Timer
+  static const uint32_t kHart = (uint32_t)kTopEarlgreyPlicTargetIbex0;
+  static const uint32_t kComparator = 0;
+  static const uint64_t kTickFreqHz =
+      1000000;  // 1 MHz so each timer tick is 1[us]
+  static const uint64_t kDeadline = UINT32_MAX;
+
+  CHECK_DIF_OK(
+      dif_rv_timer_reset(&rv_timer));  // reset the timer, disabling all IRQs,
+                                       // counters, and comparators
+
+  dif_rv_timer_tick_params_t
+      rv_timer_tick_params;  // struct to hold the timekeeping parameters for
+                             // the timer
+  // Generate an aproximate `dif_rv_timer_tick_params_t` given the device clock
+  // frequency and desired counter frequency (both given in Hertz).
+  CHECK_DIF_OK(dif_rv_timer_approximate_tick_params(
+      kClockFreqPeripheralHz  // device clock frequency
+      ,
+      kTickFreqHz  // desired counter frequency
+      ,
+      &rv_timer_tick_params  // pointer to output struct
+      ));
+  // Configure the tick params for a particular hart's counter
+  CHECK_DIF_OK(dif_rv_timer_set_tick_params(
+      &rv_timer, kHart  // hart to configure
+      ,
+      rv_timer_tick_params  // timing parameters struct
+      ));
+
+  uint64_t current_time;
+  CHECK_DIF_OK(dif_rv_timer_counter_read(&rv_timer, kHart, &current_time));
+  CHECK_DIF_OK(dif_rv_timer_arm(
+      &rv_timer, kHart, kComparator,
+      (current_time + kDeadline)  // go off when counter value is greater than
+                                  // or equal to this value
+      ));
+  CHECK_DIF_OK(
+      dif_rv_timer_counter_set_enabled(&rv_timer, kHart, kDifToggleEnabled));
+
+  LOG_INFO("SW_INFO: RV Timer active");
+
+  // ------------------------------------
+  // Alert Ping
+
+  // The ping timer will only ping alerts that have been enabled AND locked.
+  // Therefore, it should be invoked after configuring and enabling each (local)
+  // alert. The alert handler will regularly, at random intervals, ping alert
+  // sources. If a source fails to respond, a local alert will be raised. The
+  // ping timer won't start until configuration is locked.
+
+  // 1. Connfig structs for all Incoming Alerts:
+  //   (these alerts should never fire because we do not expect any incoming
+  //   alerts)
+  dif_alert_handler_alert_t alerts[ALERT_HANDLER_PARAM_N_ALERTS];
+  dif_alert_handler_class_t alert_classes[ALERT_HANDLER_PARAM_N_ALERTS];
+  for (int i = 0; i < ALERT_HANDLER_PARAM_N_ALERTS; ++i) {
+    alerts[i] = i;  // each of the alerts is added to the list
+    alert_classes[i] = kDifAlertHandlerClassA;  // configure them as Class A
+  }
+
+  // 2. Connfig structs for Local Alerts
+  // Enable "alert ping fail" local alert and configure that to classb.
```

Comment:
sorry this is a cosmetic thing, but could i ask you to place comments above the code where it's relevant instead of inline behind?  It makes it a bit hard to read with all the shifting spaces (because that's what clang-format will do).

So instead of 
```
important_code  // comment
important_code_2  // comment
```
have this instead. 
```
//comment
important_code
//comment 
important_code1
```

Reply:
Done.